rust/parquet/src/util/cursor.rs

```diff
@@ -42,20 +42,21 @@ impl SliceableCursor {
         }
     }
 
-    /// Create a slice cursor using the same data as a current one.
-    pub fn slice(&self, start: u64, length: usize) -> io::Result<Self> {
-        let new_start = self.start + start;
-        if new_start >= self.inner.len() as u64
-            || new_start as usize + length > self.inner.len()
-        {
-            return Err(Error::new(ErrorKind::InvalidInput, "out of bound"));
+    /// Create a slice cursor backed by the same data as a current one.
+    /// If the slice length is larger than the remaining bytes in the source, the slice is clamped.
```

Comment:
can we replace `clamp` with something else more plain? like "shorten"? as a non-native English speaker, I had to check dictionary for the meaning of this word :)

Reply:
I like the word but you are right 😄